messaging/WFLY-10326_messaging_global_resource_usage.adoc

```diff
@@ -0,0 +1,67 @@
+= Add Setting to Control Global Resources Usage for Messaging Servers
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10326[WFLY-10326]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-830[EAP7-830]
+* https://issues.apache.org/jira/browse/ARTEMIS-581[ARTEMIS-581]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mnovak@redhat.com[Miroslav Novak]
+
+=== Affected Projects or Components
+
+* Artemis has been providing the underlying feature since 1.4.0. It does not require a component upgrade.
+
+=== Other Interested Projects
+
+== Requirements
+
+* Add a `global-max-memory-size` attribute to the https://wildscribe.github.io/WildFly/12.0//subsystem/messaging-activemq/server/[/subsystem=messaging-activemq/server resource] to control the maximum
+amount of memory that Artemis can use to store messages for its addresses before they are considered "full" and
+their `address-full-policy` starts to apply (e.g. to drop messages, block producers, etc.). Default is `undefined` (meaning that there is no limit).
```

Comment:
This default value differs from the default one in Artemis which is Runtime.getRuntime().maxMemory() / 2.

Reply:
I'd rather use a default value of -1 instead of undefined